src/simtools/production_configuration/derive_corsika_limits.py

```diff
@@ -23,54 +24,14 @@ class LimitCalculator:
     """
 
     def __init__(self, event_data_file, telescope_list=None):
-        """
-        Initialize the LimitCalculator with the given event data file.
-
-        Parameters
-        ----------
-        event_data_file : str
-            Path to the reduced MC event data file.
-        telescope_list : list, optional
-            List of telescope IDs to filter the events (default is None).
-        """
+        """Initialize the LimitCalculator with the given event data file."""
+        self._logger = logging.getLogger(__name__)
         self.event_data_file = event_data_file
         self.telescope_list = telescope_list
-        self.event_x_core = None
-        self.event_y_core = None
-        self.simulated = None
-        self.shower_id_triggered = None
-        self.list_of_files = None
-        self.shower_sim_azimuth = None
-        self.shower_sim_altitude = None
-        self.array_azimuth = None
-        self.array_altitude = None
-        self.trigger_telescope_list_list = None
-        self.units = {}
-        self._read_event_data()
-
-    def _read_event_data(self):
-        """Read the event data from the reduced MC event data file."""
-        with tables.open_file(self.event_data_file, mode="r") as f:
-            file_names = f.root.data.file_names
-            reduced_data = f.root.data.reduced_data
-            triggered_data = f.root.data.triggered_data
-            trigger_telescope_list_list = f.root.data.trigger_telescope_list_list
-
-            self.list_of_files = file_names.col("file_names")
-
-            self.event_x_core = reduced_data.col("core_x")
-            self.event_y_core = reduced_data.col("core_y")
-            self.simulated = reduced_data.col("simulated")
-            self.shower_sim_azimuth = reduced_data.col("shower_sim_azimuth")
-            self.shower_sim_altitude = reduced_data.col("shower_sim_altitude")
-
-            self.array_altitude = triggered_data.col("array_altitudes")
-            self.array_azimuth = triggered_data.col("array_azimuths")
-            self.shower_id_triggered = triggered_data.col("shower_id_triggered")
-
-            self.trigger_telescope_list_list = [
-                [np.int16(tel) for tel in event] for event in trigger_telescope_list_list
-            ]
+
+        self.reader = SimtelIOEventDataReader(event_data_file, telescope_list=[7, 12])
```

Comment:
Why having default values here for the telescope list?

Reply:
Nono, leftover from debugging. Should be `telescope_list=telescope_list`